coderd/database/dbauthz/dbauthz.go

```diff
@@ -2234,6 +2234,15 @@ func (q *querier) GetPresetParametersByTemplateVersionID(ctx context.Context, ar
 	return q.db.GetPresetParametersByTemplateVersionID(ctx, args)
 }
 
+func (q *querier) GetPresetsAtFailureLimit(ctx context.Context, hardLimit int64) ([]database.GetPresetsAtFailureLimitRow, error) {
+	// GetPresetsAtFailureLimit returns a list of template version presets that have reached the hard failure limit.
+	// Request the same authorization permissions as GetPresetsBackoff, since the methods are similar.
+	if err := q.authorizeContext(ctx, policy.ActionViewInsights, rbac.ResourceTemplate.All()); err != nil {
```

Comment:
I still think "insights" is a weird auth context to use here.
Why did we choose this again @SasSwart?

Reply:
@dannykopping 
Basically we have 3 actions:
- `Read` Template.All()
- `Use` Template.All()
- `ViewInsights` Template.All()

`Read` is for reading the content of template
`Use` is for creating workspaces
`ViewInsights` is for gathering stats & metrics across templates

Looks like `GetPresetsBackoff` falls into `ViewInsights` category - because we gathering statistics across all template presets, calculate number of failed builds, etc...

But I'm open to suggestions

Reply:
Repurposing other RBAC roles just because they overlap logically feels like a bad design choice.
Each feature should have its own RBAC policy.

Reply:
We can introduce `new resource type` here: https://github.com/coder/coder/blob/main/coderd/rbac/object_gen.go

Then use it like this:
```go
if err := q.authorizeContext(ctx, policy.ActionViewInsights, rbac.ResourcePrebuilds.All()); err != nil {}
```

Optionally introduce new `action type` as well, and use it like this:
```go
if err := q.authorizeContext(ctx, policy.ActionCollectStatistics, rbac.ResourcePrebuilds.All()); err != nil {}
```
But I think we don't want to introduce too many action types.

cc @Emyrk

Reply:
We can do this in a follow-up, by the way 👍

Reply:
I don't have all the context for prebuilds. Can we throw a quick disc chat on the calendar?